graphql-server/src/utils/oncotree.ts

```diff
@@ -110,3 +112,50 @@ async function getOncotreeCodesFromNeo4j() {
     await session.close();
   }
 }
+
+/**
+ * Include cancer type fields in the search query indirectly by searching the Oncotree cache
+ * for matching cancerType or cancerTypeDetailed values. If a match is found, add the corresponding
+ * Oncotree code to the original search query.
+ */
+export function includeCancerTypeFieldsInSearch(
+  where: SampleWhere,
+  oncotreeCache: NodeCache
+) {
+  const customWhere = { ...where };
+
+  const sampleMetadataFilters =
+    customWhere?.hasMetadataSampleMetadata_SOME?.OR ||
+    customWhere?.OR?.find((filter) => filter.hasMetadataSampleMetadata_SOME)
+      ?.hasMetadataSampleMetadata_SOME?.OR;
+
+  if (sampleMetadataFilters?.length) {
+    const searchInput = Object.values(sampleMetadataFilters[0])[0] as
+      | string
+      | string[];
+    const searchValues = Array.isArray(searchInput)
+      ? searchInput
+      : [searchInput];
```

Comment:
Extract the user input. `searchValues` will equal `["someValue"]` for a single-value search, and `["val1", "val2", "val3", ...]` for a bulk search.